### ISSUE TITLE
Central Caching

### ISSUE THREAD

**Issue by fritzduchardt** [mykso/myks#134]
We do cache workloads, but we fetch them once for each installation. This could be improved with a central cache to pull each workload only once for each workload.

This would also allow us to add application folders in the envs directory structure, only if they really need extra configuration.

**Comment by fritzduchardt** [comment]
This is going to be a bigger rework, hence a proposal before starting to get my hands dirty:

**Aims**
- If the same helm chart and config is used for several environments, we only pull it once.
- We get rid of application-specific directories in the envs-dir structure,  if there is no specific config overwrite

**Implementation**
- We introduce a central vendir cache in the myks root with a flat hierarchy including a sub-dir for every vendir directory entry. The sub-dir has a readable name for usability plus a config digest for uniqueness:

```
.vendor
└── multus-cni-1.1.10-c5759806e4d5
```
 
- We add the application-specific directories previously held in the envs-dir structure to the `.myks' -root directory, rebuilding the envs-dir structure, but hiding it from the user:

```
.myks
├── envs
│   └── my-cluster
│       └── multus-cni
│           ├── vendir.lock.yaml
│           └── vendir.yaml
```

**Comment by Zebradil** [comment]
This sounds good and reasonable :+1:

**Comment by Zebradil** [comment]
Implemented in #274
